canonical/edgex-go
# fix(snap): update kuiper to v1.1.1 (includes ZMQ fix)

This PR updates kuiper in the snap to v1.1.1 which is the version associated with the Hanoi maintenance release (1.3.1).

## PR Checklist
Please check if your PR fulfills the following requirements:

- [NA] Tests for the changes have been added (for bug fixes / features)
- [NA] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/master/.github/Contributing.md.

## What is the current behavior?
NA

## Issue Number:
NA

## What is the new behavior?
NA

## Does this PR introduce a breaking change?



## New Imports



## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Test Instructions
1. Install mosquitto: `$ sudo snap install mosquitto`
2. Enable kuiper: `$ sudo snap set edgexfoundry kuiper=on`
3. Check logs to validate version 1.1.1 was started
4. Check that app-service-configurable and kuiper are active via snap services
 
```
  $ snap services edgexfoundry | grep app-service
  $ snap services edgexfoundry | grep kuiper
```

5. Validate that the REST API (port 9081) and client server (port 20498) are no longer accessible from another machine
6. Follow the [basic tutorial example](https://github.com/emqx/kuiper/blob/master/docs/en_US/edgex/edgex_rule_engine_tutorial.md) to ensure that kuiper is working properly:
 
Create a kuiper stream:

`$ edgexfoundry.kuiper-cli create stream demo'() WITH (FORMAT="JSON", TYPE="edgex")'
`

Create a rules file by pasting the following JSON to a file:

```
 {
  "sql": "SELECT * from demo",
  "actions": [
    {
      "mqtt": {
        "server": "tcp://broker.emqx.io:1883",
        "topic": "result",
        "clientId": "demo_001"
      }
    },
    {
      "log":{}
    }
  ]
}
```
Create the rule:

`$ edgexfoundry.kuiper-cli create rule rule1 -f ./kuiper-tutorial-rule.json 
`

Activate device-virtual:

`$ sudo snap set edgexfoundry device-virtual=on
`

Monitor the public EMQX MQTT broker:

```
$ mosquitto_sub -h broker.emqx.io -t result[{"Uint64":4162975112020145284}]
[{"Uint8":198}]
[{"Uint16":46246}]
[{"Bool":false}]
[{"Float64":-6.01462e+307}]
[{"Float32":-4.167940733580701e+36}]
[{"Int8":-74}]
[{"Int64":2799254460733675681}]
[{"Int16":4017}]
[{"Int32":-1746111203}]
[{"Bool":false}]
[{"Uint64":5927661473579216419}]
.
.
.

```
